tests/ws.c

```diff
@@ -29,15 +30,58 @@
 
 #include "testutil.h"
 
+#define SOCKET_ADDRESS  "ws://127.0.0.1:5555"
+
 /*  Basic tests for WebSocket transport. */
 
+/*  test_text() verifies that we drop messages properly when sending invalid
+    UTF-8, but not when we send valid data. */
+void test_text() {
+
+    int sb;
+    int sc;
+    int opt;
+    uint8_t bad[20];
+
+    /*  Negative testing... bad UTF-8 data for text. */
+    sb = test_socket (AF_SP, NN_PAIR);
+    sc = test_socket (AF_SP, NN_PAIR);
+
+    /*  Wait for connects to establish. */
+    nn_sleep (200);
+
+    opt = NN_WS_MSG_TYPE_TEXT;
+    test_setsockopt(sb, NN_WS, NN_WS_MSG_TYPE, &opt, sizeof (opt));
+    opt = NN_WS_MSG_TYPE_TEXT;
+    test_setsockopt(sc, NN_WS, NN_WS_MSG_TYPE, &opt, sizeof (opt));
+    opt = 500;
+    test_setsockopt(sb, NN_SOL_SOCKET, NN_RCVTIMEO, &opt, sizeof (opt));
+
+    test_bind (sb, SOCKET_ADDRESS);
+    test_connect (sc, SOCKET_ADDRESS);
+
+    test_send (sc, "GOOD");
+    test_recv (sb, "GOOD");
+
+    /*  and the bad ... */
+    strcpy((char *)bad, "BAD.");
+    bad[2] = (char)0xDD;
+    test_send (sc, (char *)bad);
+
+    /*  Make sure we dropped the frame.  Apparently it comes back as EAGAIN
+        rather than ETIMEDOUT.  I think this possibly a doc bug. */
```

Comment:
No.  The issue here is that transport layer has discarded the frame (and closed the connection).  We don't report these details back to the application, ever.

Think of this as a transport level failure, rather than an application failure.

The other thing is that other messages could reasonably arrive, and we'd never notice any error in a real world application, except that our mangled frame got dropped.  Someone would need to look at their javascript logs to see the Close message reason, I think.  That's not unique to us either -- I think all websocket implementations are probably similar here.